extensions/panache/mongodb-panache-kotlin/runtime/src/main/kotlin/io/quarkus/mongodb/panache/kotlin/reactive/ReactivePanacheMongoEntityBase.kt

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.mongodb.panache.kotlin.reactive
+
+import io.quarkus.mongodb.panache.kotlin.reactive.runtime.KotlinReactiveMongoOperations
+import io.smallrye.mutiny.Uni
+
+/**
+ * Represents an entity. If your Mongo entities extend this class they gain auto-generated accessors
+ * to all their public fields, as well as a lot of useful
+ * methods. Unless you have a custom ID strategy, you should not extend this class directly but extend
+ * [ReactivePanacheMongoEntity] instead.
+ *
+ * @see ReactivePanacheMongoEntity
+ */
+abstract class ReactivePanacheMongoEntityBase {
+    companion object {
+        val operations = KotlinReactiveMongoOperations()
+    }
```

Comment:
This was part of that reuse effort.  By making things non-static, they can be inherited and overridden etc.  If you look at the hibernate version of all this, there are some awkward contortions needed to get the kotlin versions of types in to the mix while sharing some of them.  This uses inheritance and type parameterization to greatly simplify that integration.  If this all works out, the hibernate code will be updated to work this way as well.  This approach is much "better" in my experience.

Reply:
I understand the idea but as I said, this changes an inner API that I saw some direct usage in the wild so this may not be backward compatible.
I want @FroMage advice on this, as people directly using MongoOperations or JpaOperations should not have done it so maybe we can break this.